projects/scripts/adi_board.tcl

```diff
@@ -517,27 +523,26 @@ proc ad_mem_hpx_interconnect {p_sel p_clk p_name} {
     set_property CONFIG.NUM_SI 1 $m_interconnect_cell
     ad_connect $p_rst $m_interconnect_cell/ARESETN
     ad_connect $p_clk $m_interconnect_cell/ACLK
-    ad_connect $p_rst $m_interconnect_cell/M00_ARESETN
-    ad_connect $p_clk $m_interconnect_cell/M00_ACLK
     ad_connect $m_interconnect_cell/M00_AXI $p_name_int
+    lappend sys_mem_clk_list $p_clk
     if {$p_intf_clock ne ""} {
       ad_connect $p_clk $p_intf_clock
     }
   } else {
     set_property CONFIG.NUM_SI $m_interconnect_index $m_interconnect_cell
-    ad_connect $p_rst $m_interconnect_cell/${i_str}_ARESETN
-    ad_connect $p_clk $m_interconnect_cell/${i_str}_ACLK
+    if { [lsearch $sys_mem_clk_list $p_clk] == -1} {
```

Comment:
I think we should try to check if the current clock is already connected to the smartconnect cell directly. 
We could use a command like the following:
[lsearch [get_bd_nets -of_object [get_bd_pins $m_interconenct_cell/ACLK*]] "/$p_clk"] == -1
This way we don't have to create a list, just to store the clocks.

Reply:
Added your suggestion

Reply:
Thanks, it looks good. :)